### PR TITLE
fix redundant brackets in Hive/Snowflake/Redshift

### PR DESCRIPTION
**Pull request by yuval-illumex**
Resolve https://github.com/sqlparser-rs/sqlparser-rs/issues/892

### CODE REVIEW

File: tests/sqlparser_common.rs [hunk 1]
```diff
@@ -8685,3 +8685,15 @@ fn parse_map_access_expr() {
         let _ = dialects.verified_expr(sql);
     }
 }
+
+#[test]
+fn insert_into_with_parentheses() {
+    let dialects = TestedDialects {
+        dialects: vec![
+            Box::new(SnowflakeDialect {}),
```
Comment (universalmind303) [inline tests/sqlparser_common.rs:8693]: This should work with `GenericDialect` as well.

Comment (yuval-illumex) [reply tests/sqlparser_common.rs:8693]: @universalmind303 added. 
I used Redshift + Snowflake because the issue talks about these two dialects.

Comment (alamb) [reply tests/sqlparser_common.rs:8693]: Appears it was fixed in [1fce575](https://github.com/sqlparser-rs/sqlparser-rs/pull/1229/commits/1fce575ecffaaca153849ccf55e97a375fe1421b)

**Review by alamb — APPROVED** [review]
Makes sense to me -- thanks @yuval-illumex  and @universalmind303

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Github claims this PR has conflicts that need to be resolved before merging. Can you please do so (otherwise I'll work on it when I next have a sqlparser ide open)

**Comment by yuval-illumex** [comment]
@alamb fixed, thanks!

**Comment by alamb** [comment]
Thanks again @yuval-illumex

**Pull request merged**